weave-js/src/react.tsx

```diff
@@ -499,6 +501,39 @@ export const isWeaveObjectRef = (ref: ObjectRef): ref is WeaveObjectRef => {
   return ref.scheme === 'weave';
 };
 
+const PATTERN_ENTITY = '([a-z0-9-_]+)'; // Entity name: lowercase, digits, dash, underscore
+const PATTERN_PROJECT = '([^\\#?%:]{1,128})'; // Project name
+const RE_WEAVE_OBJECT_REF_PATHNAME = new RegExp(
+  [
+    '^', // Start of the string
+    PATTERN_ENTITY,
+    '/',
+    PATTERN_PROJECT,
+    '/',
+    '(object|op)', // Weave kind
+    '/',
+    '([a-zA-Z0-9-_/. ]{1,128})', // Artifact name
```

Comment:
Would like to draw attention to this.

The wandb SDK limits artifact names.
https://github.com/wandb/wandb/blob/main/wandb/sdk/artifacts/artifact.py#L101

> Names can contain letters, numbers, underscores, hyphens, and dots.


The W&B backend is not as restrictive. It just checks <= 128 chars and allows everything but ‘/' and ‘:’ Last I checked the artifacts UI does not even validate against that. For this reason, there might be artifacts (such as Launch Jobs) whose names contain characters such as spaces that you could not create through the public W&B SDK API.

In Weave multiple external users have now published with '/' and at least one internal user has used spaces. I think we may want to be more restrictive at publish time, but do we want to allow these now to unbreak the users?

Reply:
Personally, I would rather avoid slashes entirely and stick to a more conservative dictionary for now. It feels better to intentionally add support, than have to deal with accidental ramifications.

That being said, now that you have handled this case, maybe there isn't a problem with including support for these symbols?

At the very least, we should certainly add the same restriction to the backend (and write client) to protect users from including other symbols / emojies / etc.

Reply:
We never really resolved the conversation around Python API design and whether it was OK to raise on an invalid name. https://weightsandbiases.slack.com/archives/C03BSTEBD7F/p1714676345298539

Reply:
Also we still need to discuss the pros/cons of validating at the pydantic level: https://github.com/wandb/weave/pull/1597